site/content/blog/2024-07-17-script-scan-rules/index.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "ZAP Scripts are now Full Scan Rules!"
+summary: "ZAP scripts can now do everything that scan rules can."
+images:
+- https://www.zaproxy.org/blog/2024-07-17-script-scan-rules/images/cover.jpg
+type: post
+tags:
+  - blog
+  - scanrules
+  - scripts
+date: "2024-07-17"
+authors:
+  - akshath
+---
+
+ZAP active and passive scripts are now first-class scan rules! 🎉
+
+ZAP scripts may now optionally be exposed as scan rules, allowing you to include them in your scan policies.
+
+Existing scripts are not affected by this change.
+
+## How to Expose a Script as a Scan Rule
+
+Any active or passive script that implements the `getMetadata` function will be exposed as a scan rule to ZAP.
+
+```js
+const ScanRuleMetadata = Java.type("org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata");
+
+function getMetadata() {
+	return ScanRuleMetadata.fromYaml(`
+id: 12345
+name: Passive Vulnerability Title
+description: Full description
+solution: The solution
+references:
+  - Reference 1
+  - Reference 2
```

Comment:
Make these actual URLs?

`https://example.org/1/`
Or something like that

Reply:
They don't necessary have to be URLs

Reply:
I know it's just a string but I'm pretty sure the unwritten expectation is URLs. I can't think of a scan rule where we have something else.

Reply:
I think they are always URLs tbh...

Reply:
Okay, I can update them